### PR TITLE
Santa is late but he fixed all style violations and some lint warnings

### PR DESCRIPTION
**Pull request by maxme**
This PR:

* Fixes all style violation in `WordPress/src/java`.
* Remove unused resources from the main project and promote the lint check to error.
* Fixes following lint issues
```
- warning ScrollViewSize: ScrollView size validation
- warning ApplySharedPref: Use apply() on SharedPreferences
- warning CustomViewStyleable: Mismatched Styleable/Custom View Name
- warning DuplicateIncludedIds: Duplicate ids across layouts combined with include tags
- warning InlinedApi: Using inlined constants on older versions
- warning OldTargetApi: Target SDK attribute is not targeting latest version
- warning GradleOverrides: Value overridden by Gradle build script
- warning ExtraText: Extraneous text in resource files
- warning SpUsage: Using dp instead of sp for text sizes
- warning SwitchIntDef: Missing @IntDef in Switch
- warning AddJavascriptInterface: addJavascriptInterface Called
- warning DrawAllocation: Memory allocations within drawing code
- warning Recycle: Missing recycle() calls
- warning ObsoleteLayoutParam: Obsolete layout params
- warning ObsoleteSdkInt: Obsolete SDK_INT Version Check
- warning UnusedNamespace: Unused namespace 
- warning IconDipSize: Icon density-independent size validation
- warning IconLocation: Image defined in density-independent drawable folder
```

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java [hunk 1]
```diff
@@ -29,49 +29,54 @@ public ReaderTag(String slug,
...
-        setEndpoint(endpoint);
+        setmTagDisplayName(displayName);
+        setmTagTitle(title);
+        setmEndpoint(endpoint);
         this.tagType = tagType;
     }
 
-    public String getEndpoint() {
-        return StringUtils.notNullStr(endpoint);
+    public String getmEndpoint() {
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java]: this is wrong, right? the getters and setters shouldn't contain the `m`

**Comment by maxme** [comment]
I'll use the following list to track the review this PR:

- [x] warning ScrollViewSize: ScrollView size validation.
- [x] warning ApplySharedPref: Use apply() on SharedPreferences.
- [] warning CustomViewStyleable: Mismatched Styleable/Custom View Name
- [] warning DuplicateIncludedIds: Duplicate ids across layouts combined with include tags
- [] warning InlinedApi: Using inlined constants on older versions
- [] warning OldTargetApi: Target SDK attribute is not targeting latest version
- [] warning GradleOverrides: Value overridden by Gradle build script
- [] warning ExtraText: Extraneous text in resource files
- [x] warning SpUsage: Using dp instead of sp for text sizes (I was afraid some textview were using dp on purpose to avoid breaking the layout, I tried with the biggest font size and things look ok.)
- [] warning SwitchIntDef: Missing @IntDef in Switch
- [] warning AddJavascriptInterface: addJavascriptInterface Called
- [] warning DrawAllocation: Memory allocations within drawing code
- [x] warning Recycle: Missing recycle() calls (TIL we should call `recycle()` after using a TypedArrary coming from `context.obtainStyledAttributes()`)
- [] warning ObsoleteLayoutParam: Obsolete layout params
- [] warning ObsoleteSdkInt: Obsolete SDK_INT Version Check
- [] warning UnusedNamespace: Unused namespace
- [] warning IconDipSize: Icon density-independent size validation
- [] warning IconLocation: Image defined in density-independent drawable folder

File: WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java (hunk 1 above)
Comment (maxme) [reply WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java]: Argh, yes this wrong, well spotted!

File: WordPress/src/main/java/org/wordpress/android/ui/ExpandableHeightGridView.java [hunk 2]
```diff
@@ -7,53 +7,43 @@
 
 // based on http://stackoverflow.com/questions/8481844/gridview-height-gets-cut
 
-public class ExpandableHeightGridView extends GridView
-{
-    boolean expanded = false;
+public class ExpandableHeightGridView extends GridView {
+    boolean mIsExpanded = false;
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/ExpandableHeightGridView.java]: Should the "is" be included in a variable name?

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEpilogueFragment.java [hunk 3]
```diff
@@ -64,12 +64,14 @@ protected boolean listenForLogin() {
     }
 
     @Override
-    protected @LayoutRes int getContentLayout() {
+    protected @LayoutRes
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEpilogueFragment.java]: The line break seems unnecessary, I wonder if we could make the checkstyle keep it on a single line. Feel free to ignore this comment, it just caught my eye, so I wanted to mention it.

Comment (maxme) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEpilogueFragment.java]: Yep, looks like a glitch in the AS config, I'll fix this double check the config.

File: .travis.yml [hunk 4]
```diff
@@ -34,3 +34,4 @@ install:
 
 script:
   - ./gradlew -PdisablePreDex assembleVanillaRelease lintVanillaRelease testVanillaRelease || (grep -A20 -B2 'severity="Error"' -r --include="*.xml" WordPress libs; exit 1)
+  - ./gradlew WordPress:checkstyle
```
[the remark this answers is not in the document]
Comment (maxme) [reply .travis.yml]: I'd keep the order: biggest problem should break the checks first (assemble, test, lint, checkstyle). 

No strong opinion about it, checkstyle first makes sense as well if you assume the build step will run OK after a PR has been opened.

**Comment by maxme** [comment]
> I've encountered several getters with "m" prefix and several boolean variables with "is" prefix. We might use the structural search and try to fix those. WDYT?


This should be fixed in 85f9e83

File: WordPress/src/main/java/org/wordpress/android/ui/ExpandableHeightGridView.java (hunk 2 above)
Comment (maxme) [reply WordPress/src/main/java/org/wordpress/android/ui/ExpandableHeightGridView.java]: IIRC, we added the `is`, `has` prefixes in variable and field names in FluxC to make them explicit. @aforcier do you remember if had another rationale?

File: WordPress/src/androidTest/java/org/wordpress/android/ui/posts/PostUtilsTest.java [hunk 5]
```diff
@@ -20,7 +20,7 @@ public void testCollapseShortcodes() {
     }
 
     public void testShortcodeSpaces() {
-        String postContent = "[   gallery number=\"arst\"     /]";
```
[the remark this answers is not in the document]
Comment (maxme) [reply WordPress/src/androidTest/java/org/wordpress/android/ui/posts/PostUtilsTest.java]: OOOops, that's a bad one and I actually forgot about that (I added a reminder in the tool that trim lines, but it wasn't enough).

File: WordPress/src/main/java/org/wordpress/android/ui/ExpandableHeightGridView.java (hunk 2 above)
Comment (aforcier) [reply WordPress/src/main/java/org/wordpress/android/ui/ExpandableHeightGridView.java]: Don't think there was another rationale - it's explicit and can eliminate some ambiguities (is `mPage` an object representing a page? or is it saying that the current object is a page? etc.). I don't think we actually have an enforced rule for this though.

File: WordPress/src/androidTest/java/org/wordpress/android/ui/posts/PostUtilsTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (aforcier) [reply WordPress/src/androidTest/java/org/wordpress/android/ui/posts/PostUtilsTest.java]: Was this the one mentioned here?

https://github.com/wordpress-mobile/WordPress-Android/blob/88a4386ec0c0a2882842fbf887144d93857c9cf4/tools/fix-style.sh#L28-L31

I was going to ask you about that 😀

[the remark this answers is not in the document]
Comment (maxme) [reply WordPress/src/androidTest/java/org/wordpress/android/ui/posts/PostUtilsTest.java]: Fixed in efa2a24  with a couple of other problems.

File: .travis.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (malinajirka) [reply .travis.yml]: I always try to sort tasks by duration. But it makes sense to keep the order (for now at least) as the other tasks are sorted by 'importance'.

**Comment by khaykov** [comment]
This is beautiful ✨I have only one comment - not sure about adding `m` prefix to the ViewHolder variables, considering how they are usually accessed.

**Comment by maxme** [comment]
> not sure about adding m prefix to the ViewHolder variables, considering how they are usually accessed.

**Comment by aforcier** [comment]
`OldTargetApi` is in the list but I'm not sure that's the case - this seems to be a warning that the `targetSdkVersion` is less than the most recent version in `build.gradle`, which isn't something fixed in this PR (that's a whole other project). Was this mean to be a different lint check?

**Comment by malinajirka** [comment]
>OldTargetApi is in the list but I'm not sure that's right - this seems to be a warning that the targetSdkVersion is less than the most recent version in build.gradle, which isn't something fixed in this PR (that's a whole other project). Was this mean to be a different lint check?

Lint (lintVanillaRelease) was complaining about a line of code removed in [this](https://github.com/wordpress-mobile/WordPress-Android/pull/7388/commits/7f144db25730a32e242a83423fd767688a9efb81) commit. It doesn't show any warning for the `targetSdkVersion` specified in the `build.gradle`.

**Comment by aforcier** [comment]
> Lint (lintVanillaRelease) was complaining about a line of code removed in this commit. It doesn't show any warning for the targetSdkVersion specified in the build.gradle.

Huh. When I undo that change I get the warning, but it points to the same one I get in the `build.gradle` file. I guess they've conflated the two. Anyway, makes sense 👍

File: .travis.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (maxme) [reply .travis.yml]: > line break is added after annotations in function definitions private @StringRes int method123() becomes private @StringRes /n int method123(). I'm wondering whether this is the behaviour we expect.

Fixed in af1efee

**Comment by maxme** [comment]
> line break is added after annotations in function definitions private @StringRes int method123() becomes private @StringRes /n int method123(). I'm wondering whether this is the behaviour we expect.

Fixed in af1efee

**Comment by malinajirka** [comment]
Awesome! Thanks @maxme !!!

**Comment by hypest** [comment]
I know the PR is merged (🎉) but wanted to poke your minds about this particular codestyle setting: [ALIGN_MULTILINE_PARAMETERS_IN_CALLS](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/.idea/codeStyleSettings.xml#L79)

It seems that we have more than a few code sites that are or will get "squeezed" by that. [Example](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java#L669-L687).

Any strong feelings on this one? Can we set it to `false` to give more space to the callbacks? cc @maxme

**Comment by maxme** [comment]
Right, I agree we should remove that one.

**Pull request merged**